Userland/Libraries/LibVideo/VP9/Decoder.cpp

```diff
@@ -403,6 +406,745 @@ bool Decoder::trailing_bits()
+        return m_bit_stream->read_literal(4) + 32;
+
+    auto v = m_bit_stream->read_literal(7);
+    if (v < 65)
+        return v + 64;
+    return (v << 1u) - 1 + m_bit_stream->read_literal(1);
+}
+
+u8 Decoder::inv_remap_prob(u8 delta_prob, u8 prob)
+{
+    u8 m = prob - 1;
+    auto v = inv_map_table[delta_prob];
+    if ((m << 1u) <= 255) {
+        return 1 + inv_recenter_nonneg(v, m);
+    }
+    return 255 - inv_recenter_nonneg(v, 254 - m);
+}
+
+u8 Decoder::inv_recenter_nonneg(u8 v, u8 m)
+{
+    if (v > 2 * m)
+        return v;
+    if (v & 1u)
+        return m - ((v + 1u) >> 1u);
+    return m + (v >> 1u);
+}
+
+bool Decoder::read_coef_probs()
+{
+    m_max_tx_size = tx_mode_to_biggest_tx_size[m_tx_mode];
+    for (auto tx_size = TX_4x4; tx_size <= m_max_tx_size; tx_size = static_cast<TXSize>(static_cast<int>(tx_size) + 1)) {
+        auto update_probs = m_bit_stream->read_literal(1);
+        if (update_probs == 1) {
+            for (auto i = 0; i < 2; i++) {
+                for (auto j = 0; j < 2; j++) {
+                    for (auto k = 0; k < 6; k++) {
+                        auto max_l = (k == 0) ? 3 : 6;
+                        for (auto l = 0; l < max_l; l++) {
+                            for (auto m = 0; m < 3; m++) {
+                                auto& coef_probs = m_probability_tables->coef_probs()[tx_size];
+                                coef_probs[i][j][k][l][m] = diff_update_prob(coef_probs[i][j][k][l][m]);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    return true;
+}
+
+bool Decoder::read_skip_prob()
+{
+    for (auto i = 0; i < SKIP_CONTEXTS; i++) {
+        m_probability_tables->skip_prob()[i] = diff_update_prob(m_probability_tables->skip_prob()[i]);
+    }
+    return true;
+}
+
+bool Decoder::read_inter_mode_probs()
+{
+    for (auto i = 0; i < INTER_MODE_CONTEXTS; i++) {
+        for (auto j = 0; j < INTER_MODES - 1; j++) {
+            m_probability_tables->inter_mode_probs()[i][j] = diff_update_prob(m_probability_tables->inter_mode_probs()[i][j]);
+        }
+    }
+    return true;
+}
+
+bool Decoder::read_interp_filter_probs()
+{
+    for (auto i = 0; i < INTERP_FILTER_CONTEXTS; i++) {
+        for (auto j = 0; j < SWITCHABLE_FILTERS - 1; j++) {
+            m_probability_tables->interp_filter_probs()[i][j] = diff_update_prob(m_probability_tables->interp_filter_probs()[i][j]);
+        }
+    }
+    return true;
+}
+
+bool Decoder::read_is_inter_probs()
+{
+    for (auto i = 0; i < IS_INTER_CONTEXTS; i++) {
+        m_probability_tables->is_inter_prob()[i] = diff_update_prob(m_probability_tables->is_inter_prob()[i]);
+    }
+    return true;
+}
+
+bool Decoder::frame_reference_mode()
+{
+    auto compound_reference_allowed = false;
+    for (auto i = 1; i < REFS_PER_FRAME; i++) {
+        if (m_ref_frame_sign_bias[i + 1] != m_ref_frame_sign_bias[1])
+            compound_reference_allowed = true;
+    }
```

Comment:
```suggestion
    for (size_t i = 2; i <= REFS_PER_FRAME; i++)
        if (m_ref_frame_sign_bias[i] != m_ref_frame_sign_bias[1])
            compound_reference_allowed = true;
    }
```
In these cases, I like to use `size_t` explicitly, since the vector will take `size_t` as argument. Even if you use `1u` instead of `1`, there could still be a type mismatch.

Reply:
Good call, thanks :)